docs/building_the_project_with_dpcpp.rst

```diff
@@ -0,0 +1,470 @@
+
+By default, the portBLAS backend is not tuned for any specific device.
+This tuning is required to achieve best performance.
+portBLAS can be tuned for a specific hardware target by adding compiler
+definitions in 2 ways:
+
+#.
+  Manually specify a tuning target with ``-DPORTBLAS_TUNING_TARGET=<target>``.
+  The list of portBLAS targets can be found
+  `here <https://github.com/codeplaysoftware/portBLAS#cmake-options>`_.
+  This will automatically set ``-fsycl-targets`` if needed.
+#.
+  If one target is set via ``-fsycl-targets`` the configuration step will
+  try to automatically detect the portBLAS tuning target. One can manually
+  specify ``-fsycl-targets`` via ``CMAKE_CXX_FLAGS``. See
+  `DPC++ User Manual <https://intel.github.io/llvm-docs/UsersManual.html>`_
+  for more information on ``-fsycl-targets``.
+
+portBLAS relies heavily on JIT compilation. This may cause time-outs on some
+systems. To avoid this issue, use ahead-of-time compilation through tuning
+targets or ``sycl-targets``.
+
+.. _build_for_portfft_dpcpp:
+
+Building for portFFT
+---------------------
+
+`portFFT <https://github.com/codeplaysoftware/portFFT>`_ is enabled by setting
+``-DENABLE_PORTFFT_BACKEND=True``.
+
+By default, the portFFT backend is not tuned for any specific device. The tuning
+flags are detailed in the `portFFT
+<https://github.com/codeplaysoftware/portFFT>`_ repository, and can set at
+configuration time. Note that some tuning configurations may be incompatible
+with some targets.
+
+The portFFT library is compiled using the same ``-fsycl-targets`` as specified
+by the ``CMAKE_CXX_FLAGS``. If none are found, it will compile for
+``-fsycl-targets=spir64``, and -if the compiler supports it-
+``nvptx64-nvidia-cuda``. To enable HIP targets, ``HIP_TARGETS`` must be
+specified. See `DPC++ User Manual
+<https://intel.github.io/llvm-docs/UsersManual.html>`_ for more information on
+``-fsycl-targets``.
+
+.. _build_additional_options_dpcpp:
+
+Additional Build Options
+##########################
+
+When building oneMKL the SYCL implementation can be specified by setting the
+``ONEMKL_SYCL_IMPLEMENTATION`` option. Possible values are:
+
+* ``dpc++`` (default) for the `Intel(R) oneAPI DPC++ Compiler
+  <https://software.intel.com/en-us/oneapi/dpc-compiler>`_ and for the `oneAPI
+  DPC++ Compiler <https://github.com/intel/llvm>`_ compilers.
+* ``hipsycl`` for the `AdaptiveCpp <https://github.com/illuhad/AdaptiveCpp>`_
+  SYCL implementation.
+Please see :ref:`building_the_project_with_adaptivecpp` if using this option.
+
+The following table provides details of CMake options and their default values:
+
+.. list-table::
+   :header-rows: 1
+
+   * - CMake Option
+     - Supported Values
+     - Default Value 
+   * - BUILD_SHARED_LIBS
+     - True, False
+     - True      
+   * - BUILD_DOC
+     - True, False
+     - False     
+
+
+.. note::
+  When building with clang++ for AMD backends, you must additionally set
```

Comment:
nitpicking
```suggestion
  When building with ``clang++`` for AMD backends, you must additionally set
```

Reply:
Done in [ef02116](https://github.com/oneapi-src/oneMKL/pull/510/commits/ef021162bfa1dd44b8746c8b214cf248b16cbdf9).